stream.js

```diff
@@ -121,7 +122,7 @@ Stream.prototype._resume = function () {
   if (this.cursor === -1)
     return // not ready yet
 
-  if (this.live && !this.writing && this.cursor > 0)
+  if (this.live && !this.writing && this.hasWritten)
```

Comment:
In the case of `live`, both `index.js` (when a new record is written) and a paused sink can call `stream.resume()`. In that case, this line didn't know what's the difference between those two possibilities, so when a paused sink calls `resume()` it would just keep on delivering the `0` offset record in an infinite loop. This PR introduces the boolean `hasWritten` instead of piggybacking on the value of `this.cursor`.